content/consul/v1.22.x/content/docs/hcp.mdx

```diff
@@ -0,0 +1,340 @@
+
+Export the `CONSUL_HTTP_TOKEN` environment variable in your self-managed environment and then run the following command.
+
+```shell-session
+$ consul snapshot restore /home/backup/hcp-cluster.snapshot
+Restored snapshot
+```
+
+If you cannot use use environment variables, add the `-token=` flag to the command:
+
+```shell-session
+$ consul snapshot restore /home/backup/hcp-cluster.snapshot -token="<token-value">
+Restored snapshot
+```
+
+For more information on this command, refer to the [Consul CLI documentation](/consul/commands/snapshot/restore).
+
+### Update the client configuration file to point to the new server
+
+Modify the agent configuration on your Consul clients. You must update the following configuration values:
+
+- `retry_join` IP address
+- TLS encryption
+- ACL token
+
+You can use an existing certificate authority or create a new one in your self-managed cluster. For more information, refer to [Service mesh certificate authority overview in the Consul documentation](/consul/docs/connect/ca)
+
+The following example demonstrates a modified client configuration.
+
+```hcl
+retry_join = ["<new.server.IP.address>"]
+
+tls {
+    defaults {
+        auto_encrypt {
+            allow_tls =true
+            tls = true
+        }
+        verify_incoming = true
+        verify_outgoing = true
+    }
+}
+
+acl {
+    enabled = true
+    default_policy = "deny"
+    enable_token_persistence = true
+    tokens {
+        agent = "<Token-Value>"
+    }
+}
+```
+
+For more information about configuring these fields, refer to the [agent configuration reference in the Consul documentation](/consul/docs/agent/config/config-files).
+
+### Restart the client agent and verify that the migration was successful
+
+Restart the client to apply the updated configuration and reconnect it to the new cluster.
+
+```shell-session
+$ sudo systemctl restart consul
+```
+
+After you update and restart all of the client agents, check the catalog to ensure that clients migrated successfully. You can check the Consul UI or run the following CLI command.
+
+```shell-session
+$ consul members
+```
+
+Run `consul members` on the Dedicated cluster as well. Ensure that all clients appear as `inactive` or `left`.
+
+### Disconnect supporting resources and decommission the HCP Consul Dedicated cluster
+
+After you confirm that your client agents successfully connected to the self-managed cluster, delete VPC peering connections and any other unused resources. If you use other HCP services, ensure that these resources are not currently in use. After you delete a peering connection or an HVN, it cannot be used by any HCP product.
+
+## Migrate to self-managed on Kubernetes
+
+To migrate to a self-managed Consul Enterprise cluster on Kubernetes, [connect to the Dedicated cluster's current leader node](/hcp/docs/consul/dedicated/access) and then complete the following steps.
```

Comment:
Same problem with the HCP Consul docs no longer in production. This link redirects to https://developer.hashicorp.com/hcp/docs/changelog#2025-11-12